src/NuGet.Core/NuGet.Packaging/RuntimeModel/RuntimeGraph.cs

```diff
@@ -12,8 +12,16 @@ namespace NuGet.RuntimeModel
 {
     public class RuntimeGraph : IEquatable<RuntimeGraph>
     {
-        private ConcurrentDictionary<Tuple<string, string>, bool> _areCompatible 
-            = new ConcurrentDictionary<Tuple<string, string>, bool>();
+        private readonly ConcurrentDictionary<RuntimeCompatKey, bool> _areCompatible
+            = new ConcurrentDictionary<RuntimeCompatKey, bool>();
+
+        private readonly ConcurrentDictionary<string, HashSet<string>> _expandCache
```

Comment:
Did you also measure the memory consumption for large solutions after this change?

Reply:
Haven't measured these individually, but at the end of these changes RuntimeGraphs will be shared across the solution which will reduce memory allocations. Previously it was loaded up for each project.